openpype/hosts/maya/plugins/publish/extract_yeti_rig.py

```diff
@@ -142,12 +142,22 @@ def process(self, instance):
             instance.data['transfers'] = []
 
         for resource in instance.data.get('resources', []):
-            for file in resource['files']:
-                src = file
-                dst = os.path.join(image_search_path, os.path.basename(file))
-                instance.data['transfers'].append([src, dst])
-
-                self.log.debug("adding transfer {} -> {}". format(src, dst))
+            if resource["files"]:
+                for file in resource['files']:
+                    src = file
+                    dst = os.path.join(
+                        image_search_path, os.path.basename(file))
+                    instance.data['transfers'].append([src, dst])
+            else:
+                for file in resource['source']:
+                    src = (
+                        file if os.path.isabs(file) else os.path.abspath(file)
+                    )
+                    dst = os.path.join(
+                        image_search_path, os.path.basename(file))
+                    instance.data['transfers'].append([src, dst])
```

Comment:
This would be a bug elsewhere - resources are required to have `files` defined!

The BUG here is that the collector is INCORRECTLY detecting the actual files.

The issue should thus be fixed here: https://github.com/ynput/OpenPype/pull/5990/files#diff-04814bcf9a7f695cada507547746801d136cb330dff3f47b41c2213f1698671cR152-R168

Reply:
The latest commit raises the error when there is neither texture from the texture nodes nor image search path from the yeti nodes

Reply:
I see - that means the search logic would need fixing - likely it only searches on the search paths currently - but it should also allow absolute paths.

Could you share the error?

Reply:
> I see - that means the search logic would need fixing - likely it only searches on the search paths currently - but it should also allow absolute paths.
> 
> Could you share the error?

I implement the raise PublishError in the latest commit if there is no texture in texture node and there is no image search path in the yeti nodes. Sorry for causing the misunderstanding.

Reply:
So does this mean that actual production use cases (like how one would usually work with Yeti) works fine now?

Reply:
> So does this mean that actual production use cases (like how one would usually work with Yeti) works fine now?

If there is no any bug hit when publishing Yeti rig with either image search path or texture from the texture nodesm, YES.

Reply:
Is still don't understand - what types of **bugs** are you expecting?

This is roughly what I expect to work:

**Full paths without image search paths:**
- A valid full path to texture (without image search path set)
- A valid full path to texture with <UDIM> (without image search path set)
- A valid full path to animation texture with `%04d`? (without image search path set)

**Relative paths with image search paths:**
- A relative path to texture (with image search path set)
- A relative path to texture with <UDIM> (with image search path set)
- A relative path to animation texture with `%04d`? (with image search path set)

Reply:
> Is still don't understand - what types of **bugs** are you expecting?
> 
> This is roughly what I expect to work:
> 
> **Full paths without image search paths:**
> 
> * A valid full path to texture (without image search path set)
> * A valid full path to texture with  (without image search path set)
> * A valid full path to animation texture with `%04d`? (without image search path set)
> 
> **Relative paths with image search paths:**
> 
> * A relative path to texture (with image search path set)
> * A relative path to texture with  (with image search path set)
> * A relative path to animation texture with `%04d`? (with image search path set)

It can be some other errors(not limited to syntax error, key error) coming up. I mean at least in my side, it doesn't give any errors and work as expected.